### PR TITLE
Check and improve READMEs

### LINKED ISSUE #83: Check and improve README files
**Tasks**:

- [x] Add types of Merkle trees to the `imt` package
- [x] Add circuits to the `circuits` package README file list
- [x] Add `usage` section to the `poseidon-cipher` package

### PR DESCRIPTION
**Pull request by 0xjei**
## Description
This PR aims to provide:
- An explanation of different Merkle trees implementations + an update for the usage section of the `imt` package.
- Add missing circuits to the `circuits` package README file list.
- Add `usage` section to the `poseidon-cipher` package.

## Related Issue(s)
closes #83 

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by cedoor — APPROVED** [review]
LGTM! I was thinking about some other improvements to make the README files all packages more consistent.

* `@zk-kit/smt` has a "References" section. It might make sense to use it in the other packages too when there are references.
* The "Utils" section in some packages (i.e. `@zk-kit/imt`) has a code-block/script that can copied and run on-the-fly. Others have a list of functions. I think it makes sense to follow the same structure as the `imt` package, as there's already a list of functions with their signatures (function signatures) and types in the Typedoc documentation.

What do you think?

**Comment by 0xjei** [comment]
> LGTM! I was thinking about some other improvements to make the README files all packages more consistent.

Thank you for the review :sparkle: 

> * `@zk-kit/smt` has a "References" section. It might make sense to use it in the other packages too when there are references.

Yes, makes sense. Referencing a solid foundation theory is important.

> * The "Utils" section in some packages (i.e. `@zk-kit/imt`) has a code-block/script that can copied and run on-the-fly. Others have a list of functions. I think it makes sense to follow the same structure as the `imt` package, as there's already a list of functions with their signatures (function signatures) and types in the Typedoc documentation.

Yes, I think that having an e2e example may be useful for quick integrations / showcases.

> What do you think?

I agree with everything. Do you want me to introduce this kind of changes in this PR for each package or we can punt back to another PR for now?

**Comment by cedoor** [comment]
> Do you want me to introduce this kind of changes in this PR for each package or we can punt back to another PR for now?

Great! I think this PR is fine. It's quite generic so it could include that kinds of changes.

File: packages/poseidon-proof/README.md [hunk 1]
```diff
@@ -74,57 +74,55 @@ yarn add @zk-kit/poseidon-proof
...
 ```typescript
-import { verify } from "@zk-kit/poseidon-proof"
-
-const response = await verify(fullProof)
-
-console.log(response) // true or false
-
-// Eventually you may want to check the nullifier.
+import { generate, verify } from "@zk-kit/poseidon-proof"
+;(async () => {
```
Comment (cedoor) [inline packages/poseidon-proof/README.md:79]: This is no longer necessary with the new NodeJS versions (same comment for the other packages).

`process.exit(0)` could also be omitted imo.

**Review by cedoor — CHANGES_REQUESTED** [review]

**Review by cedoor — APPROVED** [review]

**Pull request merged**